frontend/src/features/apartment/ApartmentDetailsPage.tsx

```diff
@@ -169,13 +170,26 @@ const ConfirmedPriceDetails = ({confirmed}: {confirmed: IApartmentConfirmedMaxim
     );
 };
 
-interface UnconfirmedPricesDownloadModalProps {
+interface DownloadModalProps {
     apartment: IApartmentDetails;
     isVisible: boolean;
     setIsVisible;
 }
-const UnconfirmedPricesDownloadModal = ({apartment, isVisible, setIsVisible}: UnconfirmedPricesDownloadModalProps) => {
-    const [additionalInfo, setAdditionalInfo] = useState("");
+
+const UnconfirmedPricesDownloadModal = ({apartment, isVisible, setIsVisible}: DownloadModalProps) => {
+    const formRef = useRef<HTMLFormElement>(null);
+    const downloadForm = useForm({defaultValues: {request_date: today(), additional_info: ""}});
+    const {handleSubmit} = downloadForm;
+    const handleDownloadButtonClick = () => {
+        formRef.current && formRef.current.dispatchEvent(new Event("submit", {cancelable: true, bubbles: true}));
+    };
+    const onSubmit = () => {
+        downloadApartmentUnconfirmedMaximumPricePDF(
```

Comment:
Can you add here closing the modal after the PDF is successfully downloaded?

Probably easiest with returning the `fetch` promise from `downloadApartmentUnconfirmedMaximumPricePDF` and adding a `.then` here :)